### DISCUSSION TITLE
Is it safe to echo the private key of an SSH connection to a file in GitHub Actions?

### DISCUSSION THREAD

**Discussion by KouWakai** [community/community#49370] [Actions] [answered]
Also, will the file remain on the server after the action has completed?
or does every GitHub Actions run on a new instance, making it safe?

**Comment by airtower-luna** [comment, marked as answer]
Depends on your definition of safe. :slightly_smiling_face: 

If you use GitHub hosted runners, each job gets a fresh VM that's destroyed after the job. You could always add a step with `if: always()` that deletes or even [`shred`](https://manpages.debian.org/bullseye/coreutils/shred.1.en.html)s the private key file after the job is done using it, that's basically what `actions/checkout` does with the token in its post step.

**Reply by KouWakai** [reply]
Thank you for the answer:thumbsup:
